src/graph.rs

```diff
@@ -4577,220 +4802,94 @@ impl<'a, 'graph> Builder<'a, 'graph> {
   }
 
   /// Enqueue a request to load the specifier via the loader.
-  fn load(
-    &mut self,
-    specifier: &ModuleSpecifier,
-    maybe_range: Option<&Range>,
-    is_dynamic: bool,
-    is_root: bool,
-    maybe_attribute_type: Option<AttributeTypeWithRange>,
-    maybe_version_info: Option<&JsrPackageVersionInfoExt>,
-  ) {
-    self.load_with_redirect_count(
-      0,
-      specifier,
-      maybe_range,
-      is_dynamic,
-      is_root,
-      maybe_attribute_type,
-      maybe_version_info,
-    )
+  fn load(&mut self, options: LoadOptionsRef) {
+    self.load_with_redirect_count(0, options)
   }
 
   #[allow(clippy::too_many_arguments)]
   fn load_with_redirect_count(
     &mut self,
     redirect_count: usize,
-    specifier: &ModuleSpecifier,
-    maybe_range: Option<&Range>,
-    is_dynamic: bool,
-    is_root: bool,
-    maybe_attribute_type: Option<AttributeTypeWithRange>,
-    maybe_version_info: Option<&JsrPackageVersionInfoExt>,
+    options: LoadOptionsRef,
   ) {
-    struct ProvidedModuleAnalyzer(RefCell<Option<ModuleInfo>>);
-
-    #[async_trait::async_trait(?Send)]
-    impl ModuleAnalyzer for ProvidedModuleAnalyzer {
-      async fn analyze(
-        &self,
-        _specifier: &ModuleSpecifier,
-        _source: Arc<str>,
-        _media_type: MediaType,
-      ) -> Result<ModuleInfo, JsErrorBox> {
-        Ok(self.0.borrow_mut().take().unwrap()) // will only be called once
+    let specifier = options.specifier;
+    let maybe_range = options.maybe_range;
+    let original_specifier = specifier;
+    let specifier = self.graph.redirects.get(specifier).unwrap_or(specifier);
+    if options.is_asset {
+      // this will always be set when an asset
+      if let Some(attribute) = &options.maybe_attribute_type {
+        let is_allowed = match attribute.kind.as_str() {
+          "bytes" => self.unstable_bytes_imports,
+          "text" => self.unstable_text_imports,
```

Comment:
What will happen if these unstable options are not specified and there's text or bytes import? Will it error out? Silently ignore?